x-pack/plugins/triggers_actions_ui/public/application/hooks/use_alert_data_view.ts

```diff
@@ -5,72 +5,155 @@
  * 2.0.
  */
 
-import { DataView, FieldSpec } from '@kbn/data-views-plugin/common';
+import { i18n } from '@kbn/i18n';
+import { DataView } from '@kbn/data-views-plugin/common';
 import { useKibana } from '@kbn/kibana-react-plugin/public';
-import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common';
-import type { ValidFeatureId } from '@kbn/rule-data-utils';
-import useAsync from 'react-use/lib/useAsync';
-import { useMemo } from 'react';
+import { AlertConsumers, ValidFeatureId } from '@kbn/rule-data-utils';
+import { useEffect, useMemo, useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
 import { TriggersAndActionsUiServices } from '../..';
+import { fetchAlertIndexNames } from '../lib/rule_api/alert_index';
+import { fetchAlertFields } from '../lib/rule_api/alert_fields';
 
 export interface UserAlertDataView {
-  value?: DataView[];
+  dataview?: DataView[];
   loading: boolean;
-  error?: Error;
 }
 
 export function useAlertDataView(featureIds: ValidFeatureId[]): UserAlertDataView {
-  const { http } = useKibana<TriggersAndActionsUiServices>().services;
+  const {
+    http,
+    data: dataService,
+    notifications: { toasts },
+  } = useKibana<TriggersAndActionsUiServices>().services;
+  const [dataview, setDataview] = useState<DataView[] | undefined>(undefined);
   const features = featureIds.sort().join(',');
+  const isOnlySecurity =
+    featureIds.length === 1 && (featureIds as AlertConsumers[]).includes(AlertConsumers.SIEM);
 
-  const indexNames = useAsync(async () => {
-    const { index_name: indexNamesStr } = await http.get<{ index_name: string[] }>(
-      `${BASE_RAC_ALERTS_API_PATH}/index`,
-      {
-        query: { features },
-      }
+  const hasSecurityAndO11yFeatureIds =
+    featureIds.length > 1 && (featureIds as AlertConsumers[]).includes(AlertConsumers.SIEM);
+
+  const queryIndexNameFn = () => {
+    return fetchAlertIndexNames({ http, features });
+  };
+
+  const queryAlertFieldsFn = () => {
+    return fetchAlertFields({ http, featureIds });
+  };
+
+  const onErrorFn = () => {
+    toasts.addDanger(
+      i18n.translate('xpack.triggersActionsUI.useAlertDataView.useAlertDataMessage', {
+        defaultMessage: 'Unable to load alert data view',
+      })
     );
+  };
 
-    return indexNamesStr;
-  }, [features]);
+  const {
+    data: indexNames,
+    isSuccess: isIndexNameSuccess,
+    isInitialLoading: isIndexNameInitialLoading,
+    isLoading: isIndexNameLoading,
+  } = useQuery({
+    queryKey: ['loadAlertIndexNames'],
+    queryFn: queryIndexNameFn,
+    onError: onErrorFn,
+    refetchOnWindowFocus: false,
+    enabled: featureIds.length > 0 && !hasSecurityAndO11yFeatureIds,
```

Comment:
we should make the `enabled` here clear that this hook only fetches (`loadAlertIndexNames`) or the other (`loadAlertsField`) and never both. We can achieve this by having the `enabled` mapped 1 variable, and for the other fetch use the same variable but NOT'ed if that makes sense (essentially a SIEM mode and none SIEM mode) . Trying to parse `featureIds.length > 0 && !hasSecurityAndO11yFeatureIds` and `!isOnlySecurity && !hasSecurityAndO11yFeatureIds` is kind of confusing.

Doing so might help to simplify other conditionals in this hook.